pkg/logger/logger.go

```diff
@@ -0,0 +1,85 @@
+package logger
+
+import (
+	"testing"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+// Logger is a minimal subset of smartcontractkit/chainlink/core/logger.Logger implemented by go.uber.org/zap.SugaredLogger
+type Logger interface {
+	With(args ...interface{}) Logger
```

Comment:
This self reference is a problem. Although core/logger.Logger also contains a `With(args ...interface) Logger`, the method signatures don't match. I don't think there is any way to express this with generics, but a transitional helper func might be possible, like `logger.With(Logger, args ...interface{})` :thinking:

Reply:
Added `func With`, which is not pretty, but fixes the problem. Open to better ideas :shrug:

Reply:
I know this pain. I've had to do this https://github.com/smartcontractkit/chainlink-terra/blob/develop/cmd/monitoring/main.go#L74-L82 
I think a separate function `With(args ...interface) Logger` is fine!